app/scripts/contentscript.js

```diff
@@ -412,16 +415,21 @@ const initStreams = () => {
 
 // TODO:LegacyProvider: Delete
 function getNotificationTransformStream() {
-  return createThoughStream((chunk, _, cb) => {
-    if (chunk?.name === PROVIDER) {
-      if (chunk.data?.method === 'metamask_accountsChanged') {
-        chunk.data.method = 'wallet_accountsChanged';
-        chunk.data.result = chunk.data.params;
-        delete chunk.data.params;
+  const stream = new Transform({
+    highWaterMark: 16,
+    objectMode: true,
```

Comment:
Are these the settings used by `createThoughStream` by default?

Reply:
It maps to current/previous behavior and should not constitute a change, [yes](https://github.com/rvagg/through2/blob/ebe3d0b736115ffca785336301119979df70a708/through2.js#L70).

(I hope you don't want me to answer which of all the existing constructions require which values for `highWaterMark` - which I think we shouldn't need to know to perform this change?)

Reply:
I wonder what's the purpose of choosing the highWaterMark explicitly at all. We should be fine with the defaults. Anyway, should probably stay that way for the sake of compatibility.